src/plugins/intel_cpu/src/dnnl_extension_utils.h

```diff
@@ -25,9 +26,18 @@ class Shape;
 class IMemory;
 
 class DnnlExtensionUtils {
+public:
+    struct throw_tag {};
+    struct nothrow_tag {};
```

Comment:
can't we use `std::nothrow` instead?
```c++
// which never throws because uses optional, so does not require extra args
std::optional<dnnl::memory::data_type> DnnlExtensionUtils::ElementTypeToDataTypeOpt(
    const ov::element::Type& elementType) noexcept;
// the one which throws
dnnl::memory::data_type DnnlExtensionUtils::ElementTypeToDataType(const ov::element::Type& elementType);
// the one which never throws
dnnl::memory::data_type DnnlExtensionUtils::ElementTypeToDataType(const ov::element::Type& elementType, std::nothrow) noexcept;
```

Reply:
`std::nothrow_t` was introduced to distinguish throwing and non-throwing `new` operator. I wouldn't encourage the use of this type in the `Edge` class, but it was already there. Tag dispatching look better in this context as it allows to employ static polymorphism when necessary and relatively easy to use directly.